check_limits.py

```diff
@@ -1,18 +1,61 @@
+from battery_vitals import *
 
-def battery_is_ok(temperature, soc, charge_rate):
-  if temperature < 0 or temperature > 45:
-    print('Temperature is out of range!')
-    return False
-  elif soc < 20 or soc > 80:
-    print('State of Charge is out of range!')
-    return False
-  elif charge_rate > 0.8:
-    print('Charge rate is out of range!')
-    return False
 
-  return True
+def battery_is_ok(battery_validators):
+    for battery_validator in battery_validators:
+        value_state = battery_validator.get_value_state()
+        if value_state != ValueState.Normal:
+            return False, battery_validator, value_state
+    return True, None, ValueState.Normal
 
 
 if __name__ == '__main__':
-  assert(battery_is_ok(25, 70, 0.7) is True)
-  assert(battery_is_ok(50, 85, 0) is False)
+    result, validator, state = battery_is_ok([BatteryTemperature(40)])
+    assert (result is True)
+    assert (validator is None)
+    assert (state == ValueState.Normal)
+
+    result, validator, state = battery_is_ok([BatteryTemperature(100, Unit.Fahrenheit)])
```

Comment:
Good selection of tests to cover the combinations of breach and the metric.
Another approach would be to test the base class (`BatteryVital`) in all the 5 conditions. And then simply verify that the threshold-numbers supplied to the different classes are as per the requirement.